packages/nx/src/utils/project-graph-utils.ts

```diff
@@ -22,9 +23,28 @@ export function projectHasTargetAndConfiguration(
   configuration: string
 ) {
   return (
-    projectHasTarget(project, target) &&
-    project.data.targets[target].configurations &&
-    project.data.targets[target].configurations[configuration]
+    // Explicitly defined target + configuration
+    (projectHasTarget(project, target) &&
+      project.data.targets[target].configurations &&
+      project.data.targets[target].configurations[configuration]) ||
+    // Inferred configuration from presence of .env files
+    (projectHasTarget(project, target) &&
+      [
+        join(workspaceRoot, `.env.${configuration}`),
```

Comment:
I don't think having a file there should define a configuration. Configurations should still be defined in the project configuration. If we'd like an easier way to define configurations, that is a separate discussion.

Can we remove this part from the changes here and consider them separately?

Reply:
> I don't think having a file there should define a configuration. Configurations should still be defined in the project configuration. If we'd like an easier way to define configurations, that is a separate discussion.

Removing the requirement to redefine a configuration under every target is what myself and other customers are trying to get. With the current state of environment files in NX, a great deal of technical debt and configuration overhead is generated making it an unusable solution for most projects ultimately making NX very limiting.